test/release/examples/primers/loops.chpl

```diff
@@ -0,0 +1,950 @@
+  If there is no explicit synchronization between the iterations,
+  perhaps a forall loop would be a better choice, since it would use a
+  number of tasks proportional to the targeted hardware parallelism?
+
+
+  Closing Discussions
+  ===================
+
+  At this point, you've learned about all of Chapel's loop forms.  The
+  remaining sections cover some loop-related discussion topics that
+  come up frequently in practice.
+
+
+  Nesting Loops
+  -------------
+
+  The loop forms discussed here can be nested arbitrarily, and their
+  definitions are the same whether they are an outer or inner loop.  A
+  nested for-loop will perform nested serial iterations as in other
+  languages.  A nested coforall-loop will create a number of tasks
+  equal to outer loop's trip count and the sum of all the inner loops'
+  counts.  For example, the following loop will logically create
+  ~`x**2` tasks, since each iteration of both loops will create its
+  own task.
+
+*/
+
+  config const x = 4;
+
+  writeln();
+  coforall i in 1..x do
+    coforall j in 1..x do
+      writeln("Here's a message from one of the nested coforall tasks");
+
+/*
+
+  A tricky case to reason about in a nested loop situation is the
+  forall-loop since its implementation is essentially "Do whatever the
+  parallel iterator says to."  If a parallel iterator were to always
+  create ``x`` tasks (say), then a nested forall loop that invoked
+  that iterator for both loops would create ``x**2`` tasks as in the
+  coforall example above.
+
+  In practice, however, most parallel iterators (including those
+  defined on ranges, domains, and arrays) will take stock of the
+  number of tasks running on the current locale and then throttle the
+  number of tasks they create to avoid overwhelming the node.  As a
+  result, a nested forall-loop over a pair of ranges, like:
+  
+*/
+
+  forall i in 1..500 do
+    forall j in 1..500 do
+      /* do some computation here */;
+
+/*      
+
+  will typically create only ``numCores`` tasks in total.
+  Specifically, the outer loop will create ``numCores`` tasks, then
+  each of the inner loops will see that all the cores are busy and
+  avoid creating additional tasks to avoid creating needless
+  parallelism.
+  
+  In such cases, if the inner loop body was not computationally
+  intensive, it could make sense to rewrite the inner loop as a
+  ``foreach`` in order to avoid the overhead of having the iterator
+  determine whether or not to create tasks at all:
+
+*/
+
+  forall i in 1..500 do
+    foreach j in 1..500 do
+      /* do some computation here */;
+
+/*
+
+  That said, such overheads are relatively modest, so loop bodies that
+  are computationally intense, the benefit for changing the inner loop
+  from ``forall`` to ``foreach`` may be negligible.
+
+  In summary, there is nothing magical about nested loops.  When
+  reasoning about what a given loop nest does, simply consider the
```

Comment:
I would recommend leaving out the word "simply".  It can be inferred to be saying that the reader should get this because this stuff is simple, but if the reader is a new Chapel programmer they might not think it is simple and then think they shouldn't even be trying if they don't get it right away.